src/main/main.cpp

```diff
@@ -3,529 +3,386 @@
 
-static void
-repairCorruptedTx(
-    boost::asio::steady_timer& timer,
-    clio::Config const& config,
-    Backend::CassandraBackend& backend,
-    ripple::uint256 const& hash)
-{
-    clio::LogService::info() << " - repairing " << hash;
-    auto const data = requestFromRippled(
-        timer,
-        config,
-        {
-            {"method", "tx"},
-            {"transaction", to_string(hash)},
-            {"binary", true},
-        });
+                if (tx.ledgerSequence > ledgerRange.maxSequence)
+                    continue;
 
-    maybeWriteTransaction(backend, data);
-}
+                try
+                {
+                    ripple::STTx const sttx{ripple::SerialIter{
+                        tx.transaction.data(), tx.transaction.size()}};
+                    if (sttx.getTxnType() != ripple::TxType::ttNFTOKEN_MINT)
+                        continue;
+
+                    ripple::TxMeta const txMeta{
+                        sttx.getTransactionID(),
+                        tx.ledgerSequence,
+                        tx.metadata};
+                    toWrite.push_back(
+                        std::get<1>(getNFTDataFromTx(txMeta, sttx)).value());
+                }
+                catch (std::exception const& e)
+                {
+                    clio::LogService::warn()
+                        << "Corrupted tx detected: " << hash;
+                    std::cerr << "Corrupted tx detected: " << hash << std::endl;
+
+                    if (not repairAddress_.has_value())
+                    {
+                        clio::LogService::fatal()
+                            << "Not attempting to repair. Rerun with `-repair "
+                               "[host:port]` to repair corrupted transactions.";
+                        exit(-1);
+                    }
+
+                    repairCorruptedTx(timer_, *repairAddress_, backend_, hash);
+
+                    auto maybeTx = backend_->fetchTransaction(hash, yield);
+
+                    if (!maybeTx.has_value())
+                    {
+                        clio::LogService::fatal()
+                            << "Could not fetch written transaction for hash "
+                            << hash << "; Repair failed.";
+                        exit(-1);
+                    }
+
+                    txs[idx] = maybeTx.value();
+                    --idx;  // repeat the try section for the repaired tx
+                    std::cerr << "+ tx repaired: " << hash << std::endl;
+                }
+            }
 
-static std::vector<NFTsData>
-doNFTWrite(
-    std::vector<NFTsData>& nfts,
-    Backend::CassandraBackend& backend,
-    std::string const& tag)
-{
-    auto const size = nfts.size();
-    if (size == 0)
-        return nfts;
-    backend.writeNFTs(std::move(nfts));
-    backend.sync();
-    clio::LogService::info() << tag << ": Wrote " << size << " records";
-    return {};
-}
+            toWrite = maybeDoNFTWrite(toWrite, backend_, tag_);
```

Comment:
maybeDoNFTWrite not really write to db. If it does not write through db, toWrite in memory can not be resumed.